### DISCUSSION TITLE
Filter template Variable from JSON Query using Infinity datasource

### DISCUSSION THREAD

**Discussion by Rodan07** [grafana/grafana-infinity-datasource#265] [Q&A]
I would like to filter out some of the results from a JSON query before the variable is passed to the rest of grafana. When I try to use the filter option it never has any fields to filter on. I have only selected two columns so that it acts as a key:value pair. I have tried using the Regex filter but it appears to only filter out based on the value and not the key, the value is a random serial number and the key is a model number, I would like to filter out the model numbers that are not relevant. Unfortunately the OpenAPI I am using does not offer any filtering.

**Comment by yesoreyeram** [comment]
Can you provide some more info if possible? Like screenshots of the query editor, sample data,  what version of plugin and grafana you are using..

Hard to tell without more details. Sorry for the inconveniences you had.

**Reply by Rodan07** [reply]
Of course, thank you for the reply. 
Grafana v 8.4.1
Infinity v 0.8.0

From the image I am trying to filter out all of the MR and MS models but as you can see it is filtering on the value part of the key:value pair. It filtered out any serial that didn't have a Z or a MX in it. 

Here is the variable as a key:value pair.
![grafanaVariableRegex](https://user-images.githubusercontent.com/29843675/155408281-33f8b529-7e63-47fa-a897-88250b0ee67d.jpg)

Here is the same variable with just the serial column so you can see the only serials left have a Z in them.
![grafanaModelremoved](https://user-images.githubusercontent.com/29843675/155410854-232b0ad3-f764-4d7a-8004-0196d134dcd8.jpg)

Here is the built in filter with showing it give no columns to filter on. 
![grafanaVariableFilter](https://user-images.githubusercontent.com/29843675/155408293-9cbde3da-bcb4-4811-b845-cae9b8bc0fe2.jpg)

Here is some sample data.

[
    {
        "serial": "Q2SW-P9GY-GBHU",
        "networkId": "L_568016503002104305",
        "tags": [],
        "name": "MX01-01",
        "model": "MX250",
        "floorPlanId": null
    },
    {
        "serial": "Q2KD-HWZL-243U",
        "networkId": "L_568016503002104305",
        "name": "AP01-07",
        "model": "MR42",
        "floorPlanId": "g_568016503002104291"
    },
    {
        "serial": "Q2KD-FWR3-5UE7",
        "networkId": "L_568016503002104305",
        "name": "AP01-06",
        "model": "MR42",
        "floorPlanId": "g_568016503002104291"
    },
    {
        "serial": "Q2WD-S4BS-PEEU",
        "networkId": "L_568016503002104305",
        "name": "AP01-04",
        "model": "MR42",
        "floorPlanId": "g_568016503002104291"
    },
    {
        "serial": "Q2KD-XQVE-9W6M",
        "networkId": "L_568016503002104305",
        "name": "AP01-03",
        "model": "Z42",
        "floorPlanId": "g_568016503002104291"
    }
    
]

**Reply by yesoreyeram** [reply]
Thank you for the details. Definitely seems to be a bug. Fixing soon.

**Reply by Rodan07** [reply]
Thank you for letting me know. 👍

**Reply by emabiz** [reply]
I have the same problem. I tried also with transformation, but it seems that variables are not supported in transformations.
Then it become fundamental to have a way to filter data dynamically.
Do you have an ETA for the fix?
Could I help in some way?
